### PR TITLE
spectra: init at 1.0.1

### PR DESCRIPTION
**Pull request by Vonfry**
###### Description of changes
A new package [spectra](https://spectralib.org/) on nixpkgs.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/spectra/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    sha256 = "sha256-HaJmMo4jYmO/j53/nHrL3bvdQMAvp4Nuhhe8Yc7pL88=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  propagatedBuildInputs = [ eigen ];
+
+  meta = with lib; {
+    homepage = "https://spectralib.org/";
+    description = "Spectra stands for Sparse Eigenvalue Computation Toolkit as a Redesigned ARPACK. It is a C++ library for large scale eigenvalue problems, built on top of Eigen, an open source linear algebra library.";
```
Comment (smancill) [inline pkgs/development/libraries/spectra/default.nix:25]:
```suggestion
    description = "A C++ library for large scale eigenvalue problems, built on top of Eigen";
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -35593,6 +35593,8 @@ with pkgs;
 
   refind = callPackage ../tools/bootloaders/refind { };
 
+  spectra = callPackage ../development/libraries/spectra {};
```
Comment (smancill) [inline pkgs/top-level/all-packages.nix:35596]:
```suggestion
  spectra = callPackage ../development/libraries/spectra { };
```

File: pkgs/development/libraries/spectra/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    sha256 = "sha256-HaJmMo4jYmO/j53/nHrL3bvdQMAvp4Nuhhe8Yc7pL88=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  propagatedBuildInputs = [ eigen ];
+
+  meta = with lib; {
+    homepage = "https://spectralib.org/";
+    description = "A C++ library for large scale eigenvalue problems, built on top of Eigen.";
```
Comment (smancill) [inline pkgs/development/libraries/spectra/default.nix:25]:
```suggestion
    description = "A C++ library for large scale eigenvalue problems, built on top of Eigen";
```

It must be without the dot, as most other packages.

**Review by smancill — APPROVED** [review]

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 187056` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spectra</li>
  </ul>
</details>

**Review by rapenne-s — APPROVED** [review]

**Pull request merged**